### PR TITLE
Ticket 5688: Newport controller

### PR DESCRIPTION
**Pull request by DominicOram**
### Description of work

Created IOC for the Newport controller. Documentation at https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Newport. I added no tests as there is no emulator so any tests would basically only be testing against simulated axes, which we already test against in the galil tests.

### To test

* Run up in DEVSIM and confirm it creates simulated axes
* Ask me for connection details for the real hardware (will probably need to be done with someone in the office, I suspect a 30 minute ZOOM of "If I press this button does it move" is sufficient)
---

#### Code Review

- [ ] A copy of the manual has been placed on the shared drive
- [ ] Pertitent information has been stored in the [wiki](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Specific-Device-IOC)
- [ ] Does the IOC conform to IBEX standards?
    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)?
- [ ] Do the IOC and support module conform to the new [build guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reducing-Build-Dependencies)
- [ ] Have the changes been recorded appropriately in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?
- [ ] Is the device's flow control setting correct? [For most devices flow control should be OFF](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Flow-control).

### Functional Tests

- IOC responds correctly in:
    - [ ] Devsim mode
    - [ ] Recsim mode
    - [ ] Real device, if available
- [ ] Supplementary IOCs (`..._0n` where `n>1`) run correctly
- [ ] Log files do not report undefined macros (serach for `macLib: macro` to find instances of `macLib: macro [macro name] is undefined...`

### Final steps

- [ ] Update the IOC submodule in the main EPICS repo. See [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

**Review by JamesKingWork — CHANGES_REQUESTED** [review]
I think the motorUtils is not being initialised correctly and a couple of minor comments and questions

File: NWPRTXPS/NWPRTXPS-IOC-01App/src/build.mak [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+$(APPNAME)_DBD += asyn.dbd 
+
+# Add all the support libraries needed by this IOC
+## ISIS standard libraries ##
+$(APPNAME)_LIBS += asubFunctions
+$(APPNAME)_LIBS += devIocStats 
+$(APPNAME)_LIBS += pvdump $(MYSQLLIB) easySQLite sqlite 
+$(APPNAME)_LIBS += caPutLog
+$(APPNAME)_LIBS += icpconfig pugixml
+$(APPNAME)_LIBS += calc autosave
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply NWPRTXPS/NWPRTXPS-IOC-01App/src/build.mak:50]: The calc dbd is `calcSupport.dbd`, which I am including.

File: NWPRTXPS/iocBoot/iocNWPRTXPS-IOC-01/st-common.cmd [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# specify additional directories in which to to search for included request files
+set_requestfile_path("${MOTOR}/motorApp/Db", "")
+## as we are common, we need to explicity define the 01 area for when we are run by 02, 03 etc 
+set_requestfile_path("${TOP}/iocBoot/iocNWPRTXPS-IOC-01", "")
+
+iocshCmdLoop("< st-ctrl.cmd", "CNUM=\$(I)", "I", 1, 24)
+iocshCmdLoop("< st-max-axis.cmd", "MN=\$(I)", "I", 1, 4)
+
+# Make sure controller number is 2 digits long
+calc("MTRCTRL", "$(MTRCTRL=11)", 2, 2)
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply NWPRTXPS/iocBoot/iocNWPRTXPS-IOC-01/st-common.cmd:15]: I've removed the default entirely. I think any default will only lead to confusion.

File: NWPRTXPS/iocBoot/iocNWPRTXPS-IOC-01/st-common.cmd [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# configure axes
+< $(NEWPORTCONFIG)/axes.cmd
+
+# motion set points etc.
+< $(NEWPORTCONFIG)/motionsetpoints.cmd
+< $(NEWPORTCONFIG)/sampleChanger.cmd
+< $(NEWPORTCONFIG)/motorExtensions.cmd
+
+## motor util package
+dbLoadRecords("$(MOTOR)/db/motorUtil.db","P=$(MYPVPREFIX)$(IOCNAME):,$(IFIOC)= ,PVPREFIX=$(MYPVPREFIX)")
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply NWPRTXPS/iocBoot/iocNWPRTXPS-IOC-01/st-common.cmd:45]: Good spot!

[the remark this answers is not in the document]
Comment (DominicOram) [reply NWPRTXPS/iocBoot/iocNWPRTXPS-IOC-01/st-common.cmd:45]: https://github.com/ISISComputingGroup/EPICS-motor/pull/36

**Review by JamesKingWork — APPROVED** [review]

**Pull request merged**